language/Type-Classes.md

```diff
@@ -137,36 +137,128 @@ See also the section in [PureScript by Example](https://book.purescript.org/chap
 
 ## Type Class Deriving
 
-Some type class instances can be derived automatically by the PureScript compiler. To derive a type class instance, use the `derive instance` keywords:
+The compiler can derive type class instances to spare you the tedium of writing boilerplate. There are a few ways to do this depending on the specific Type and Class being derived.
 
-```purescript
-newtype Person = Person { name :: String, age :: Int }
+### Derive from `newtype`
```

Comment:
I think newtype deriving should come after regular deriving. It's really common to throw together a sum type and then derive `Eq` and `Ord` instances, for example, and I think that's a nice motivating example for deriving in general.

Newtype deriving is a bit special because a) you can derive more things and b) it has some subtle behavior around instances using the same implementation as the underlying type that deserves an explanation.

What do you think of shifting this to after `derive` has been introduced?